hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/KeyOutputStream.java

```diff
@@ -205,8 +209,10 @@ public synchronized void write(byte[] b, int off, int len)
     if (len == 0) {
       return;
     }
-    handleWrite(b, off, len, false);
-    writeOffset += len;
+    synchronized (this) {
+      handleWrite(b, off, len, false);
```

Comment:
handleWrite() calls allocateBlockIfNeeded() which means it's going to get blocked waiting for OM response, which is fine. It doesn't happen frequent so shouldn't be a big deal. We can revisit later if it turns out to be not the case.

Reply:
Yes that is correct.